scripts/next_step_assignment.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+from EPPs.common import step_argparser, StepEPP
+from EPPs.config import load_config
+# from pyclarity_lims.entities import ProtocolStep
+from pyclarity_lims.entities import Protocol
+
+
+class AssignNextStep(
+    StepEPP):  # finds the next step in the protocol and assigns it to the artifacts appearing in the Next Steps of the script -
+    # assumes the next step wanted is the next step in the protocol i.e. doesn't skip one or more steps in the configuration
+    # assumes that all artifacts should have the same next step
+
+    def _run(self):
+        current_step = self.process.step.configuration  # configuration gives the ProtocolStep entity.
+
+        protocol = Protocol(self.process.lims, uri='/'.join(self.process.step.configuration.uri.split('/')[:-2]))
+
+        actions = self.process.step.actions  # obtain the actions of the step # creates a StepActions entity for the current step
+
+        next_actions = actions.next_actions  # obtain the next actions in the step #creates a list of dict for next_actions for the step
+
+        steps = protocol.steps  # a list of all the ProtocolSteps in protocol
+
+        # print(steps)
+
+        for step in steps:  # find the index of the current step in the list of all ProtocolSteps
+            if step == current_step:
```

Comment:
Are you able here to just get the last step (steps[-1]) and see if that == current step?